### PR TITLE
Added support for Python classpath isolation

### PR DESCRIPTION
**Pull request by genadi-hp**
This includes only infrastructure support: having all maven artifacts on local filesystem and descriptor file prepared.

Sign-off: Genadi Rabinovich, genadi@hpe.com

### CODE REVIEW

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/MavenConfigImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public static final String MAVEN_REPO_LOCAL = "cloudslang.maven.repo.local";
+    public static final String USER_HOME        = "user.home";
+    @Override
+    public String getLocalMavenRepoPath() {
+        String defValue = System.getProperty(USER_HOME) + SEPARATOR + ".m2" + SEPARATOR + "repository";
+        return System.getProperty(MAVEN_REPO_LOCAL, defValue);
+    }
+
+    @Override
+    public String getRemoteMaveRepoUrl() {
```
Comment (Bonczidai) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/MavenConfigImpl.java]: typo: Mave**n**

File: dependency-management/dependency-management-impl/src/test/java/io.cloudslang.dependency.impl.services/DependencyServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = DependencyServiceTest.TestConfig.class)
+public class DependencyServiceTest {
+    @Autowired
+    private DependencyService dependencyService;
+
+    @Test
+    public void testDependencyResolution() {
+        Set<String> ret = dependencyService.getDependencies(new HashSet<>(Arrays.asList("groupId1:test-artifact:1.0",
```
Comment (Bonczidai) [inline dependency-management/dependency-management-impl/src/test/java/io.cloudslang.dependency.impl.services/DependencyServiceTest.java]: maybe break this test case in two

File: runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/PythonExecutionCachedEngine.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Set;
+
+/**
+ * Created by Genadi Rabinovich, genadi@hpe.com on 05/05/2016.
+ */
+public class PythonExecutionCachedEngine extends ExecutionCachedEngine<PythonExecutor> implements PythonExecutionEngine{
+    @Autowired
+    private DependencyService dependencyService;
+
+    @Value("#{systemProperties['python.executor.cache.size'] != null ? systemProperties['python.executor.cache.size'] : 100}")
```
Comment (Bonczidai) [inline runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/PythonExecutionCachedEngine.java]: IMHO opinion we should not use hardcoded values in the new modules, rather extract them as constants for example

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/MavenConfigImpl.java (hunk 1 above)
Comment (genadi-hp) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/MavenConfigImpl.java]: Ok, thanks

File: dependency-management/dependency-management-impl/src/test/java/io.cloudslang.dependency.impl.services/DependencyServiceTest.java (hunk 2 above)
Comment (genadi-hp) [inline dependency-management/dependency-management-impl/src/test/java/io.cloudslang.dependency.impl.services/DependencyServiceTest.java]: I have split first test into two.

**Pull request merged**